### DISCUSSION TITLE
FASTAPI BackgroundTasks

### DISCUSSION THREAD

**Discussion by MounikaAB** [fastapi/fastapi#11929] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
def start_background_task(self, background_tasks, user_email, job_id):
        background_tasks.add_task(
            self.insert_data_in_ioid_tables, user_email, job_id
        )

    def load_data_from_table(self, background_tasks, user_email):
        job_id = self.get_status(user_email)
        self.start_background_task(background_tasks, user_email, job_id)
        return f"Succesfully started job with jobid {job_id}"
```

### Description

Trying to use add_task function in background_tasks to run it in background(asynchronous way). But it is taking 1-2 hrs approx to complete the logic in asynchronous way. Whereas same code is taking 150sec approx in synchronous way to complete without background_tasks.
 Is there any option to add priority to the background task or any other solution to make the execution faster.
 
 Note: We cannot run parallel calls as we are changing the same database table.
 

### Operating System

Linux

### Operating System Details

Docker image hosted on GCP. Created using the official Python base image 3.11

### FastAPI Version

0.103.2

### Pydantic Version

2.4.2

### Python Version

3.11

**Comment by rvishruth** [comment]
Hey @MounikaAB! Few thoughts here:

>It’s taking approximately 1-2 hours to complete the logic in an asynchronous manner, whereas the same code takes around 150 seconds to complete synchronously without background_tasks.

This seems unusual to me. I would suggest investigating what's causing this. It might be helpful for other readers if you could post a self-contained minimum reproducible example here. Additionally, could you clarify how you're calculating the 'total time to run'? Are you measuring from the time the background task was initiated or from when the request was sent?

>Note: We cannot run parallel calls as we are modifying the same database table.

Even if you're updating the same database table, parallel calls are still possible. I'm not sure which database you're using, but for example, with Postgres, I recommend reading this: [PostgreSQL Transaction Isolation Levels](https://www.postgresql.org/docs/current/transaction-iso.html).

Depending on the database you're using, there might be other options as well.

**Comment by YuriiMotov** [comment]
Looks like the author is not interested in this anymore. 
Let's close this for now to keep the discussions clean.
If you’ve found a solution, please share it here.
If you still need help, feel free to let us know.
